### DISCUSSION TITLE
How come that OOM occurs later in training although start_by_longest is used

### DISCUSSION THREAD

**Discussion by Ca-ressemble-a-du-fake** [coqui-ai/TTS#2038] [General Q&A]
Hi,

I run the vits training recipe with start_by_longest=True but it fails because of OOM at step 3325.

I thought start_by_longest was OOM proof. Why doesn't it 100% protect against it ?
Is there a way I can be sure OOM won't be triggered during training ?

Thanks in advance for your suggestions,

**Comment by neurlang** [comment]
you can't be sure because the batches are shuffled presumably. It would be good to shuffle deterministically.

**Reply by Ca-ressemble-a-du-fake** [reply]
Thank you @neurlang. Now that I have decreased batch group size to 0 I don't get any OOM. Is it somehow linked to your explanation ?

**Comment by erogol** [comment]
start_by_longest is not OOM proof it only loads the largest batch at the beginning of the training so you don't wait hours for an OOM error.

**Reply by Ca-ressemble-a-du-fake** [reply]
Thank you @erogol. This is what I thought it was! OOM appeared after a couple of hours, should it have happened much earlier or is it within the expected time range ?
